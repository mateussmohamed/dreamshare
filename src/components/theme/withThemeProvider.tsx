import React, { Component, FC } from 'react'
import { ThemeProvider } from 'styled-components'

import theme from './index'

const getDisplayName = (component: FC) =>
  component.displayName || component.name || 'Component'

const withThemeProvider = (WrappedComponent: FC) =>
  class extends Component {
    static displayName = `withThemeProvider(${getDisplayName(
      WrappedComponent,
    )})`

    render() {
      return (
        <ThemeProvider theme={theme}>
          <WrappedComponent {...this.props} />
        </ThemeProvider>
      )
    }
  }

export default withThemeProvider
