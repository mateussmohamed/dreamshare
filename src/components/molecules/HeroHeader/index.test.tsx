import React from 'react'

import { HeroHeader } from './index'
import withThemeProvider from '../../theme/withThemeProvider'

describe('[<HeroHeader />]', () => {
  test('matches to snapshot', () => {
    const wrapper = withThemeProvider(() => <HeroHeader />)

    expect(wrapper).toMatchSnapshot()
  })
})
