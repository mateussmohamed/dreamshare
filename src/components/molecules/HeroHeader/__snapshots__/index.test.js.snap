// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<HeroHeader />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "HeroHeader__HeroHeaderContainer-sc-5xwzmq-0",
        "isStatic": true,
        "rules": Array [
          "justify-content:space-between;align-items:center;position:relative;z-index:2;margin-left:0;margin-right:0;@media only screen and (min-width:320px){margin-top:20px;flex-direction:column-reverse;}@media only screen and (min-width:768px){flex-direction:row;}@media only screen and (min-width:1024px){margin-top:40px;}",
        ],
      },
      "displayName": "HeroHeader__HeroHeaderContainer",
      "render": [Function],
      "styledComponentId": "HeroHeader__HeroHeaderContainer-sc-5xwzmq-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
>
  <Column
    flex="none"
    mt={
      Array [
        100,
        0,
        0,
      ]
    }
  >
    <Logo
      color="#fff"
      size={20}
      text="DREAMSHARE"
      weight="bold"
    />
  </Column>
  <Column
    flex="none"
  >
    <Button
      ghost={true}
    >
      Login
    </Button>
    <Button>
      Sign Up
    </Button>
  </Column>
</StyledComponent>
`;
